pjsip/src/pjsip/sip_transport.c

```diff
@@ -1653,6 +1653,11 @@ PJ_DEF(pj_ssize_t) pjsip_tpmgr_receive_packet( pjsip_tpmgr *mgr,
 		    mgr->on_rx_msg(mgr->endpt, PJSIP_ERXOVERFLOW, rdata);
 		    /* Exhaust all data. */
 		    return rdata->pkt_info.len;
+     		} else if (msg_status == PJSIP_EMISSINGHDR) {
+        	    /* No content-length header which is not allowed. */
+        	    PJ_LOG(2,(THIS_FILE, 
+                              "No content-length header in TCP packet"));
+        	    return -1;
```

Comment:
I don't think that "-1" is defined as the error return anywhere.  There are some other error checks but they are all on (size_eaten < 0) so I think we can change this to -PJSIP_EMISSINGHDR which will help differentiate the cause of hte error.